### PR TITLE
PLT-1367: Update logic to renew keycloak client

### PR DESCRIPTION
**Pull request by n5nk**
## Change description
Remove state management of token from multiple places. And move everything to one place where token is created. Also updated checks for race condition during token creation and expiry.

Keycloak client doesnt renew auth token after expiry
- As keycloak token gets expires in a day, there are checks which check if token is expired, and tries to refresh.
- A miss in the code, ignored the token refresh, causing 401 unauthorized error.

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java [hunk 1]
```diff
@@ -65,11 +64,11 @@ private OkHttpClient getOkHttpClient() {
     };
 
     public String getAuthToken() {
-        if (Objects.nonNull(currentAccessToken)) {
+        if (!isTokenExpired()) {
             return currentAccessToken.getToken();
         }
         synchronized (this) {
-            if (Objects.isNull(currentAccessToken)) {
+            if (isTokenExpired()) {
```
Comment (nikhilbonte21) [inline client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java:71]: remove if, execution will reach here only if isTokenExpired() == true, no need of checking again

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java [hunk 2]
```diff
@@ -89,14 +88,12 @@ public String getAuthToken() {
     }
 
     public boolean isTokenExpired() {
-        return (currentTime() + DEFAULT_MIN_VALIDITY_MINS) >= expirationTime;
```
Comment (nikhilbonte21) [inline client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java:92]: why DEFAULT_MIN_VALIDITY_MINS is removed?

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java (hunk 1 above)
Comment (n5nk) [reply client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java:71]: there can be multiple threads, which does this checks outside mutex, its like safety net, not to create another token.

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java (hunk 2 above)
Comment (n5nk) [reply client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java:92]: its not required, we are taking care in  during `expirationTime`  itself. We are doing -10mins and setting the expiry time.

**Pull request merged**